tensorrt_llm/_torch/auto_deploy/transform/library/fuse_moe.py

```diff
@@ -0,0 +1,543 @@
+                w3_list = expert_weights["w3"]
+
+                if weight_type == "fp8":
+                    fused_moe_node = graph.call_function(
+                        torch.ops.auto_deploy.torch_quant_fp8_moe,
+                        args=(
+                            hidden_states,
+                            selected_experts,
+                            normalized_routing_weights,
+                            w1_list,
+                            w2_list,
+                            w3_list,
+                            expert_scales["w1_input_scale"],
+                            expert_scales["w2_input_scale"],
+                            expert_scales["w3_input_scale"],
+                            expert_scales["w1_weight_scale"],
+                            expert_scales["w2_weight_scale"],
+                            expert_scales["w3_weight_scale"],
+                        ),
+                    )
+                elif weight_type == "fp4":
+                    fused_moe_node = graph.call_function(
+                        torch.ops.auto_deploy.torch_quant_fp4_moe,
+                        args=(
+                            hidden_states,
+                            selected_experts,
+                            normalized_routing_weights,
+                            w1_list,
+                            w2_list,
+                            w3_list,
+                            expert_scales["w1_input_scale"],
+                            expert_scales["w2_input_scale"],
+                            expert_scales["w3_input_scale"],
+                            expert_scales["w1_weight_scale"],
+                            expert_scales["w2_weight_scale"],
+                            expert_scales["w3_weight_scale"],
+                            expert_scales["w1_alpha"],
+                            expert_scales["w2_alpha"],
+                            expert_scales["w3_alpha"],
+                        ),
+                    )
+                else:
+                    fused_moe_node = graph.call_function(
+                        torch.ops.auto_deploy.torch_moe,
+                        args=(
+                            hidden_states,
+                            selected_experts,
+                            normalized_routing_weights,
+                            w1_list,
+                            w2_list,
+                            w3_list,
+                        ),
+                    )
+
+            final_hidden_state_node.replace_all_uses_with(fused_moe_node)
+            graph.erase_node(final_hidden_state_node)
+
+            while _remove_dead_inplace_nodes_in_region(gm.graph, start_boundary, end_boundary):
+                gm.graph.eliminate_dead_code()
+
+            num_moe_patterns += 1
+
+        canonicalize_graph(gm)
+
+        ad_logger.info(f"Found {num_moe_patterns} MoE Patterns")
+        ad_logger.debug("After MoE Pattern Matching: " + str(gm))
```

Comment:
also remove this, see other comment. I won't call out all the places where you can remove this but I think you get the idea. Please remove it in other places as well